src/javaRosa.js

```diff
@@ -837,6 +838,34 @@ define([
             });
 
         if (options.path === 'labelItext') {
+            if (EXPERIMENTAL_UI) {
+                $input.atwho({
+                    at: "#form",
+                    data: _.chain(mug.form.getMugList())
+                           .map(function(mug) {
+                                return {
+                                    id: mug.ufid,
+                                    name: mug.form.getAbsolutePath(mug, true),
+                                    path: mug.absolutePath
+                                };
+                            })
+                            .filter(function(choice) { return choice.name; })
+                            .value(),
+                    displayTpl: '<li>${name}</li>',
+                    insertTpl: '<output value="${path}" />',
+                    limit: 100,
+                    maxLen: 30,
+                    callbacks: {
+                        matcher: function(flag, subtext) {
```

Comment:
That's understandable I had to dig around some of their code to figure it out exactly.

The first part comes from here: https://github.com/ichord/At.js/wiki/Base-Document#custom-query-matcher

subtext is the entire string before the cursor. It then looks for either the start of file or leading whitespace then the flag. Then it returns whatever was filter the user input.

> also, nitpicky, but \w includes underscores
I did that as it makes the regex slightly clearer to me. It'd also be cool if there was some fuzzy searching that allowed _ to be changed to -. Might look into this as a nice to have feature, but thought it would be nice to get some early user feedback and put it under an experimental flag